### PR TITLE
Wrap `AdjointGenerator` in namespaces

### PR DESCRIPTION
**Pull request by tgymnich**
Removes `using namespace llvm` from header

### CODE REVIEW

File: enzyme/Enzyme/AdjointGenerator.h [hunk 1]
```diff
@@ -42,8 +42,13 @@
 #include "TypeAnalysis/TBAA.h"
 
 #define DEBUG_TYPE "enzyme"
+
+namespace ADGenLLVM {
+
 using namespace llvm;
```
Comment (wsmoses) [inline enzyme/Enzyme/AdjointGenerator.h:48]: Can you try to put the using namespace llvm inside of the `class AdjointGenerator` instead?

E.g.

```
template <class AugmentedReturnType = AugmentedReturn *>
class AdjointGenerator
    : public llvm::InstVisitor<AdjointGenerator<AugmentedReturnType>> {
private:
using namespace llvm;
```

or is that not legal?

Comment (tgymnich) [reply enzyme/Enzyme/AdjointGenerator.h:48]: using namespace is not allowed on  a class level

Comment (wsmoses) [reply enzyme/Enzyme/AdjointGenerator.h:48]: I feel like ADGen is too specific if it's just going to contain this.

What if we just had the enzyme namespace or an anonymous namespace

Comment (tgymnich) [reply enzyme/Enzyme/AdjointGenerator.h:48]: It cannot be anonymous since we need to refer to it in our outer `using`

Comment (tgymnich) [reply enzyme/Enzyme/AdjointGenerator.h:48]: change names to `Enzyme` and `EnzymeLLVM`

Other options I can see are:
- Go through the whole thing an namespace all llvm symbols.
- Put the whole thing in a .cpp implementation file and implicitly instantiate the instances of the template we will use
- `using llvm::%symbol%` all the llvm symbols we use on a class level.

Comment (wsmoses) [reply enzyme/Enzyme/AdjointGenerator.h:48]: Would this not work?

```
wmoses@beast:~/git/Enzyme/enzyme ((HEAD detached at origin/main)) $ git diff
diff --git a/enzyme/Enzyme/AdjointGenerator.h b/enzyme/Enzyme/AdjointGenerator.h
index 32018269ad2..7da33d37222 100644
--- a/enzyme/Enzyme/AdjointGenerator.h
+++ b/enzyme/Enzyme/AdjointGenerator.h
@@ -42,6 +42,9 @@
 #include "TypeAnalysis/TBAA.h"
 
 #define DEBUG_TYPE "enzyme"
+
+namespace {
+
 using namespace llvm;
 
 // Helper instruction visitor that generates adjoints
@@ -12810,3 +12813,5 @@ public:
                                         subretused);
   }
 };
+
+}
```

**Pull request closed without merging**